Makefile

```diff
@@ -68,13 +68,17 @@ uninstall:
 	rm -rf $(INSTALLBASE)/$(INSTALLNAME)
 
 restart-shell:
-	echo "Restart shell!"
+	@echo "Restart shell!"
+ifneq ($(WAYLAND_DISPLAY),) # Don't restart if WAYLAND_DISPLAY is set
```

Comment:
I got this check from https://unix.stackexchange.com/a/559950. Unfortunately there's a lot of different ways to detect Wayland but this seemed the simplest.

Reply:
I would have gone for `$XDG_SESSION_TYPE` out of instinct, but reading through the linked StackExchange post, I agree that using `$WAYLAND_DISPLAY` makes sense (the former apparently doesn't work in some environments without a display manager).